metricbeat/module/system/process_summary/process_summary.go

```diff
@@ -0,0 +1,102 @@
+// +build darwin freebsd linux windows
+
+package process_summary
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
+	"github.com/elastic/beats/metricbeat/module/system/process"
+
+	sigar "github.com/elastic/gosigar"
+)
+
+// init registers the MetricSet with the central registry.
+// The New method will be called after the setup of the module and before starting to fetch data
+func init() {
+	if err := mb.Registry.AddMetricSet("system", "process_summary", New, parse.EmptyHostParser); err != nil {
+		panic(err)
+	}
+}
+
+// MetricSet type defines all fields of the MetricSet
+// As a minimum it must inherit the mb.BaseMetricSet fields, but can be extended with
+// additional entries. These variables can be used to persist data or configuration between
+// multiple fetch calls.
+type MetricSet struct {
+	mb.BaseMetricSet
+}
+
+// New create a new instance of the MetricSet
+// Part of new is also setting up the configuration by processing additional
+// configuration entries if needed.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+
+	config := struct{}{}
+
+	if err := base.Module().UnpackConfig(&config); err != nil {
+		return nil, err
+	}
+
+	return &MetricSet{
+		BaseMetricSet: base,
+	}, nil
+}
+
+// Fetch methods implements the data gathering and data conversion to the right format
+// It returns the event which is then forward to the output. In case of an error, a
+// descriptive error must be returned.
+func (m *MetricSet) Fetch() (common.MapStr, error) {
+
+	pids, err := process.Pids()
+	if err != nil {
+		logp.Warn("Getting the list of pids: %v", err)
+		return nil, err
+	}
+
+	var summary struct {
+		sleeping int
+		running  int
+		idle     int
+		stopped  int
+		zombie   int
+	}
+
+	for _, pid := range pids {
+		state := sigar.ProcState{}
+		err = state.Get(pid)
+		if err != nil {
+			return nil, fmt.Errorf("Error getting process state for pid=%d: %v", pid, err)
+		}
+
+		switch byte(state.State) {
+		case 'S':
+			summary.sleeping += 1
+		case 'R':
+			summary.running += 1
+		case 'D':
+			summary.idle += 1
+		case 'T':
+			summary.stopped += 1
+		case 'Z':
+			summary.zombie += 1
+		default:
+			logp.Err("Unknown state <%v> for process with pid %d", state.State, pid)
```

Comment:
This could be another reason for having an `unknown` state in the summary. Without it, the sum of the counts of each state will not equal the `total` in this case.